### PR TITLE
TST, MAINT: module name excluded in typing tests

### LINKED ISSUE #18942: Allow module-names to be excluded in the typing tests
The type comments in the typing [`reveal`](https://github.com/numpy/numpy/tree/main/numpy/typing/tests/data/reveal) tests are currently rather verbose, the main contributor 
being the fact that all type names are currently prepended with their respective module, 
as this is how mypy reports them. 

Ideally we can strip the module names from the reported types, allowing us to cut down on the verbosity by a lot.
This should be especially advantageous when classes are located in sub-modules.

Examples
----------
Before:
``` python
reveal_type(divmod(f8, AR_b))  # E: Tuple[numpy.ndarray[Any, numpy.dtype[numpy.floating[Any]]], numpy.ndarray[Any, numpy.dtype[numpy.floating[Any]]]]
```

After (ideally):
``` python
reveal_type(divmod(f8, AR_b))  # E: Tuple[ndarray[Any, dtype[floating[Any]]], ndarray[Any, dtype[floating[Any]]]]
```

### PR DESCRIPTION
**Pull request by sayantikabanik**
### About PR
Issue reference - [18942](https://github.com/numpy/numpy/issues/18942)
excluded module name from the reported types

### CODE REVIEW

**Comment by sayantikabanik** [comment]
```python

### Error
Expected reveal: '47: note: Revealed type is "Tuple[numpy.ndarray[Any, numpy.dtype[Any]], numpy.ndarray[Any, numpy.dtype[Any]]]"'
2021-11-03T10:50:25.3126974Z E           Observed reveal: 'Tuple[ndarray[Any, dtype[Any]], ndarray[Any, dtype[Any]]]

### Tracing back 
    if reveal not in expected_reveal:
        raise AssertionError(
            _REVEAL_MSG.format(lineno, expression, expected_reveal, reveal)
        )
```
`reveal` is constructed using the RE, while modifying the `expected_reveal` could solve this issue
tracing further I landed with `ParameterSet` under `structures.py`

After which I am lost, could use some guidance

**Comment by BvB93** [comment]
For this to work you'll also have to strip all modules from the mypy output, otherwise the comments 
in the test-files won't match its output. You could use the regex pattern below:

``` python
import re


def strip_func(match: re.Match[str]) -> str:
    """`re.sub` helper function for stripping module names."""
    return match.groups()[1]


# Regex pattern for stripping module-names
STRIP_PATTERN = re.pattern(r"(\w+\.)+(\w+)")

```

Followed by an update to [test_typing.py::_test_reveal](https://github.com/numpy/numpy/blob/bc087bb2b3104547b097e03e669fb5fa77b16f05/numpy/typing/tests/test_typing.py#L368-L379):
``` diff
-     if reveal not in expected_reveal:
+     stripped_reveal = STRIP_PATTERN.sub(strip_func, reveal)
+     stripped_expected_reveal = STRIP_PATTERN.sub(strip_func, expected_reveal)
+     if stripped_reveal not in stripped_expected:
```

File: numpy/typing/tests/test_typing.py [hunk 1]
```diff
@@ -373,9 +378,15 @@ def _test_reveal(
     lineno: int,
 ) -> None:
     """Error-reporting helper function for `test_reveal`."""
-    if reveal not in expected_reveal:
+    strip_pattern = re.pattern(r"(\w+\.)+(\w+)")
```
Comment (BvB93) [inline numpy/typing/tests/test_typing.py:381]:
```suggestion
    strip_pattern = re.compile(r"(\w+\.)+(\w+)")
```
Ah, the function is called `compile`, not `pattern`.

Comment (sayantikabanik) [reply numpy/typing/tests/test_typing.py]: thank you

**Comment by BvB93** [comment]
Thanks @sayantikabanik.

**Pull request merged**